plan/physical.go

```diff
@@ -88,7 +88,17 @@ func removeYields(plan *PlanSpec) (*PlanSpec, error) {
 			}
 
 			newRoot := root.Predecessors()[0]
-			newRoot.RemoveSuccessor(root)
+			newSucc := make([]PlanNode, 0, len(newRoot.Successors()))
+
+			for _, succ := range newRoot.Successors() {
+				newSucc = append(newSucc, succ)
+				if succ == root {
```

Comment:
Don't you want to place this check before you add `succ` to `newSucc`?

Reply:
This is embarrassing.